File: .claude/api docs/on_provizyon_yeni/Python/capture.py


# Mağaza bilgileri
merchant_id   = 'MAGAZA_NO';
merchant_key  = b'XXXXXXXXXXX';
merchant_salt = b'YYYYYYYYYYY';

# Kullanıcı bilgileri
user_ip='';

// İşlem bilgileri

payment_amount = '14.55';
reference_no   = 'AXXXXXXXXXXXXXXXX';
merchant_oid   = 'TESTPREAUTHXXX';
client_lang    = 'tr';
test_mode      = '0';

# paytr_token

 hash_str = merchant_id + user_ip + merchant_oid +  reference_no + payment_amount
paytr_token = base64.b64encode(hmac.new(merchant_key, hash_str.encode() + merchant_salt, hashlib.sha256).digest())

 context = {
 
        'auth_type':'capture',
        'merchant_id': merchant_id,
        'client_ip': user_ip,
        'merchant_oid': merchant_oid,
        'payment_amount': payment_amount,
        'reference_no':reference_no,
        'test_failed': test_mode,
        'paytr_token': paytr_token.decode(),
        'client_lang': client_lang
    }


 return render(request, 'home.html', context)

File: .claude/api docs/on_provizyon_yeni/PHP/capture.php
<?php

// Mağaza bilgileri
$merchant_id   = 'MAGAZA_NO';
$merchant_key  = 'XXXXXXXXXXX';
$merchant_salt = 'YYYYYYYYYYY';

// Kullanıcı bilgileri
if( isset( $_SERVER["HTTP_CLIENT_IP"] ) ) {
    $ip = $_SERVER["HTTP_CLIENT_IP"];
} elseif( isset( $_SERVER["HTTP_X_FORWARDED_FOR"] ) ) {
    $ip = $_SERVER["HTTP_X_FORWARDED_FOR"];
} else {
    $ip = $_SERVER["REMOTE_ADDR"];
}

// İşlem bilgileri
$payment_amount = '14.55';
$reference_no   = 'AXXXXXXXXXXXXXXXX';
$merchant_oid   = 'TESTPREAUTHXXX';
$client_lang    = 'tr';
$test_mode      = 0;

// paytr_token
$hash_str    = $merchant_id . $ip . $merchant_oid . $reference_no . $payment_amount;
$paytr_token = base64_encode(hash_hmac('sha256', $hash_str . $merchant_salt, $merchant_key, TRUE));

$payload = [
    'auth_type'      => 'capture',
    'merchant_id'    => $merchant_id,
    'paytr_token'    => $paytr_token,
    'merchant_oid'   => $merchant_oid,
    'reference_no'   => $reference_no,
    'payment_amount' => $payment_amount,
    'client_ip'      => $ip,
    'client_lang'    => $client_lang,
    'test_failed'    => $test_mode
];

$ch = curl_init();
curl_setopt($ch, CURLOPT_URL, 'https://www.paytr.com/odeme/auth');
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
curl_setopt($ch, CURLOPT_FRESH_CONNECT, TRUE);
curl_setopt($ch, CURLOPT_TIMEOUT, 20);

$result = @curl_exec($ch);
curl_close($ch);

echo $result;